bridge/src/xcm_asset_transactor.rs

```diff
@@ -0,0 +1,81 @@
+use core::marker::PhantomData;
+
+use codec::Encode;
+use xcm::latest::{Junction, MultiAsset, MultiLocation, XcmContext};
+use xcm::prelude::*;
+use xcm_executor::{Assets, traits::TransactAsset};
+use hex_literal::hex;
+use sygma_traits::{AssetTypeIdentifier, TransactorForwarder};
+
+pub struct XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder>(PhantomData<(CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder)>);
+
+impl<CurrencyTransactor: TransactAsset, FungiblesTransactor: TransactAsset, AssetTypeChecker: AssetTypeIdentifier, Forwarder: TransactorForwarder> TransactAsset for XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder> {
+    // deposit_asset implements the TransactAsset deposit_asset method and contains the logic to classify
+    // the asset recipient location:
+    // 1. recipient is on the local parachain
+    // 2. recipient is on non-substrate chain(evm, cosmos, etc.)
+    // 3. recipient is on the remote parachain
+    fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
+        match (who.parents, who.first_interior()) {
+            // 1. recipient is on the local parachain
+            (0, Some(Parachain(_))) => {
```

Comment:
According to the XCM implementation, when someone sends from another parachain, they generally set location like  (1, [Parachain(xx), ...], but notably, the location will be converted to (0, [...]), saying that the `local` will not contain `parents` and `Parachain` junction.

So here if send to local, the pattern should be `(0, account)`